tools/check_links.py

```diff
@@ -31,37 +32,57 @@
 LINK = collections.namedtuple('LINK', 'dest valid')
 
 
-def check_docs(dir_name):
-  'Traverse dir_name and check links in Markdown files.'
+def check_link(link, readme_path, external):
+  'Checks if a link element has a valid destination.'
+  link_valid = None
+  url = urllib.parse.urlparse(link.dest)
+  # If the link is public, say the link is anyway valid
+  # if --external is not set; check the link otherwise
+  if url.scheme:
+    link_valid = True
```

Comment:
nit I would return early for legibility
```python
if not external:
  return True
```

similar for the try/except, so that you don't need the `link_valid` variable